File: 프로그래머스/lv2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    for(let i = 1; i <= yellow; i++){
        if(yellow % i !== 0) continue;
        const x = i;
        const y = yellow / i;
        if(brown === x * 2 + y * 2 + 4) return x > y ? [x + 2, y + 2] : [y + 2, x + 2];
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int totalSquares = brown + yellow;
        
        for (int width = 3; width <= Math.sqrt(totalSquares); width++) {
            if (totalSquares % width == 0) {
                int height = totalSquares / width;
                if ((width -2) * (height - 2) == yellow) {
                    return new int[]{height, width};
                }
            }
        }
        
        return new int[]{-1,-1};
    }
}